### ISSUE TITLE
Grant Admin-Consent using a Service Principal

### ISSUE THREAD

**Issue by v-perib** [Azure/azure-cli#11159]
**Describe the bug**
Customer is not able to use the admin_consent in AZ CLI: 
https://docs.microsoft.com/en-us/cli/azure/ad/app/permission?view=azure-cli-latest#az-ad-app-permission-admin-consent 

Customer is trying to automate the process and instead of using a user account (with admin privileges), he's using a service principal (with admin privileges) for signing in, but when he tries to grant admin consent with this service principal an error returns:

	"AADSTS50058: A silent sign-in request was sent but no user is signed in."

**To Reproduce**
Login in Azure Cli using a Service Principal with Application Administrator privileges
Run command to grant admin consent to an applicaiton: 

	az ad app permission admin-consent --id 00000000-0000-0000-0000-000000000000

**Expected behavior**
We could see that when customer tries to perform steps above an OAuth2 OBO flow Call is executed: 
https://docs.microsoft.com/en-us/azure/active-directory/develop/v1-oauth2-on-behalf-of-flow

However, since the OAuth 2.0 On-Behalf-Of (OBO) flow enables an application that invokes a service or web API to pass user authentication to another service or web API, i.e., the OBO flow propagates the delegated user identity and permissions through the request chain, and we have no user identity, but rather a service principal identity, the flow does not proceed since there is no valid user. 

So basically we would like to verify the following:

a) Is AZ CLI aware that for flows that use Service Principals the OAuth2 OBO flow will fail?
If true, can you update the documentation accordingly? (https://docs.microsoft.com/en-us/cli/azure/ad/app/permission?view=azure-cli-latest#az-ad-app-permission-admin-consent)

b) If false,are you able to evaluate a change on the product to use a different auth flow that allows customers to use Service Principals for sign-in and automation?

Thank you.

**Comment by fengzhou-msft** [comment]
@jiasli please take a look.

**Comment by ejfinkel** [comment]
I have also run into this issue while trying to automate the creation of an AKS cluster and the associated service principals through Terraform. I have it run the azure cli command above to grant admin consent, and this fails when I run it on Jenkins, which logs in using a service principal.

I was able to recreate it locally with the following two commands:
```
$ az login --service-principal -u "<REDACTED>" -p "<REDACTED>" --tenant "<REDACTED>"
$ az ad app permission admin-consent --id "<REDACTED>"
Bad Request({"ClassName":"Microsoft.Portal.Framework.Exceptions.ClientException","Message":"AADSTS50058: A silent sign-in request was sent but no user is signed in ...
```

Due to this bug, there does not seem to be a way to automate creating a service principal with admin consent.

**Comment by acotrez** [comment]
I run into the same issue trying to automate AKS deployment. 
Is there any roadmap for a fix ?

**Comment by vinodhbasavani** [comment]
I am also facing the same issue. Is there any work around for this issue?

**Comment by jiasli** [comment]
Same issue as #8912, #10403 

According to https://github.com/Azure/azure-cli/issues/8912#issuecomment-501404372, this is currently not supported by AAD. We are working internally with AAD team on this. Stay tuned!

**Comment by rbankole** [comment]
> I am also facing the same issue. Is there any work around for this issue?

only workaround it to use a **user account** on azure as opposed to a service principal.  ie. create a jenkins user on Azure

**Comment by jiasli** [comment]
Please check https://github.com/Azure/azure-cli/issues/12137 and see if you can get unblocked.
